pkg/machines/libvirt-dbus.js

```diff
@@ -800,11 +804,46 @@ const LIBVIRT_DBUS_PROVIDER = {
                             dispatch(updateVm(Object.assign({}, props, dumpxmlParams)));
                         else
                             dispatch(updateOrAddVm(Object.assign({}, props, dumpxmlParams)));
+
+                        dispatch(getVmSnapshots({ connectionName, domainPath: objPath }));
                     })
                     .catch(ex => console.warn("GET_VM action failed for path", objPath, ex.toString()));
         };
     },
 
+    GET_VM_SNAPSHOTS({ connectionName, domainPath }) {
+        return dispatch => call(connectionName, domainPath, 'org.libvirt.Domain', 'ListDomainSnapshots', [0], { timeout, type: 'u' })
+                .then((objPaths) => {
+                    const snaps = [];
+                    const promises = [];
+
+                    objPaths[0].forEach(objPath => promises.push(call(connectionName, objPath, 'org.libvirt.DomainSnapshot', 'GetXMLDesc', [0], { timeout, type: 'u' })));
+
+                    // WA to avoid Promise.all() fail-fast behavior
+                    const toResultObject = (promise) => {
+                        return promise
+                                .then(result => ({ success: true, result }))
+                                .catch(error => ({ success: false, error }));
+                    };
+
+                    Promise.all(promises.map(toResultObject)).then(snapXmlList => {
```

Comment:
If one of the calls fails the next promises will not run.
This is not what you want here I guess.

Reply:
I think this is the approach we use for fetching a list of everything (VMs, storage pools, virtual networks...).
What could be alternative?

Also what are the chances this is gonna fail on one snapshot but not on the other, since it's same operation done over the list of snapshots provided by libvirt API?

Reply:
You need to call catch in all promises inside promise all to avoid that. (And that's what we do in other resource types)

I guess with snapshots it's harder that one fails parsing only, but with VMs (since their configuration is much more complext) it has happened multiple times.

Reply:
Right, thanks